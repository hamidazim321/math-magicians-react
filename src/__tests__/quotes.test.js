import React from 'react';
import { render, screen } from '@testing-library/react';
import Quotes from '../components/quotes'; // Make sure the path is correct

describe('Render a quote or error', () => {
  test('renders quote', async () => {
    const response = [
      {
        quote: 'mocked quote',
      },
    ];
  
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue(response),
    });
  
    render(<Quotes />);
    
    await screen.findByText(/mocked quote/);
  
    const quotesDiv = document.getElementsByClassName('quotes')[0]
  
    expect(quotesDiv).toMatchSnapshot()
  });
  
  test('show error when the quote is not there', async ()=>{
    global.fetch = jest.fn().mockResolvedValue({
      json: jest.fn().mockResolvedValue(new Error()),
    });
  
    render(<Quotes />);
  
    expect((await screen.findByText(/Error/)).textContent).toBe('Error loading the quotes. Please try again.')
  })

})