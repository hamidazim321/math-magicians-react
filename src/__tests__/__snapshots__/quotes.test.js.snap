// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a quote or error renders quote 1`] = `
<div
  class="quotes"
>
  <q
    class="quote"
  >
    mocked quote
  </q>
</div>
`;

exports[`renders quote 1`] = `
<div
  class="quotes"
>
  <q
    class="quote"
  >
    mocked quote
  </q>
</div>
`;
